src/robotFactory.js

```diff
@@ -1,10 +1,83 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, { x = 0, y = 0 }, chipVersion) {
```

Comment:
that's not obvious what param is passed over constructor
```suggestion
  constructor(name, weight, coords = { x = 0, y = 0 }, chipVersion) {
```